src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotheap/HighHeapUsageYoungGenRca.java

```diff
@@ -118,6 +139,111 @@ protected void remove(SlidingWindowData e) {
     this(rcaPeriod, 1.0, heap_Used, gc_Collection_Time);
   }
 
+  private boolean fullGcTimeTooHigh(double avgFullGcTime) {
+    return (!Double.isNaN(avgFullGcTime) && avgFullGcTime > FULL_GC_TIME_THRES_MS);
+  }
+
+  private boolean promotionRateTooHigh(double avgPromotionRate, double modifier) {
+    return (!Double.isNaN(avgPromotionRate) && avgPromotionRate > promotionRateThreshold * modifier);
+  }
+
+  private boolean youngGcTimeTooHigh(double avgYoungGCTime) {
+    return (!Double.isNaN(avgYoungGCTime) && avgYoungGCTime > youngGenGcTimeThreshold);
+  }
+
+  private boolean prematurePromotionTooHigh(double avgGarbagePromoted) {
+    return (!Double.isNaN(avgGarbagePromoted)
+        && avgGarbagePromoted <= 1
+        && avgGarbagePromoted > garbagePromotionPctThreshold);
+  }
+
+  private double getFollowerCheckTimeoutMs() {
+    return FULL_GC_TIME_THRES_MS;
+  }
+
+  /**
+   * Creates and returns a ResourceFlowUnit that reflects the health of this RCA
+   *
+   * @param avgPromotionRate the average rate of promotion of objects from the young generation to
+   *                         the old generation in MB/s
+   * @param avgYoungGCTime the average amount of time spent on GC in ms
+   * @param avgGarbagePromoted the average amount of objects which were promoted to the old generation
+   *                           then collected from the old generation shortly after. This is a percentage
+   *                           from 0.0 to 1.0
+   * @param avgFullGCTime The average amount of time spend of full GC in ms
+   * @return A ResourceFlowUnit encapsulating the health of this RCA
+   */
+  private ResourceFlowUnit<HotResourceSummary> computeFlowUnit(double avgPromotionRate,
+                                                               double avgYoungGCTime,
+                                                               double avgGarbagePromoted,
+                                                               double avgFullGCTime) {
+    ResourceContext context = new ResourceContext(State.UNHEALTHY);
+    HotResourceSummary summary = null;
+    boolean unhealthy = true;
+
+    // Check if the RCA is unhealthy
+    if (fullGcTimeTooHigh(avgFullGCTime)) {
+      summary = new HotResourceSummary(FULL_GC_PAUSE_TIME, getFollowerCheckTimeoutMs(), avgFullGCTime,
```

Comment:
This is too complex of a change for this diff. Currently FlowUnit contains a Union of various summary types. One way to do this would be to create a new CompoundResourceSummary type that tracks issue counts across various resources with a Map and add that into the FlowUnit union.

However, this is a structural change and I'd like to avoid making it here. As is, I believe this diff evaluates conditions in a reasonable conservative order.